src/clr-angular/utils/popover/position-operators.spec.ts

```diff
@@ -16,10 +16,57 @@ import {
   flipSidesAndNudgeContent,
   nudgeContent,
   testVisibility,
+  positionsEqual,
+  posiblePositionsAsStrings,
+  transposeStringToPosition,
+  transposePositionToString,
+  SIMPLE_TO_SMART_POSITIONS,
+  validPosition,
 } from './position-operators';
 import { ClrPopoverContentOffset } from './interfaces/popover-content-offset.interface';
 import { ClrViewportViolation } from './enums/viewport-violation.enum';
 
+export function ClrPositionHelperSpec(): void {
+  describe('ClrPosition helper functions', () => {
+    it('should return array of posible positions', () => {
+      expect(posiblePositionsAsStrings()).toEqual(Object.keys(SIMPLE_TO_SMART_POSITIONS));
+    });
+
+    it('should let you find position object by passing string names', () => {
+      posiblePositionsAsStrings().forEach(position => {
+        expect(transposeStringToPosition(position)).toEqual(SIMPLE_TO_SMART_POSITIONS[position]);
+      });
+
+      expect(transposeStringToPosition('NOT_VALID_NAME')).toEqual(null);
+    });
+
+    it('should find string name of position by passing position object', () => {
+      posiblePositionsAsStrings().forEach(position => {
+        expect(position).toBe(transposePositionToString(SIMPLE_TO_SMART_POSITIONS[position]));
+      });
+
+      // Not a valid position
+      expect(null).toBe(transposePositionToString({ axis: -5, side: -5, anchor: -5, content: -5 }));
+    });
+
+    it('should check that the position is valid', () => {
+      expect(validPosition('bottom-right')).toBe(true);
+      expect(validPosition('NOT_VALID_NAME')).toBe(false);
+    });
+
+    it("should compare two position and return false if they don't match", () => {
```

Comment:
It would be nice to try with randomly ordered properties. The current implementation should be okay, but some future optimized comparison function might fail.